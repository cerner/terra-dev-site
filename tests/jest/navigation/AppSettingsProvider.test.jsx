import React from 'react';
import AppSettingsProvider from '../../../src/navigation/_AppSettingsProvider';

describe('ThemeContextProvider', () => {
  describe('Snapshots', () => {
    it('should render with no config', () => {
      const config = {};
      const wrapper = shallow((
        <AppSettingsProvider settingsConfig={config}>
          <div />
        </AppSettingsProvider>
      ));

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with default config specified', () => {
      const config = {
        defaultLocale: 'es',
        defaultTheme: 'my theme',
        defaultDirection: 'rtl',
      };
      const wrapper = shallow((
        <AppSettingsProvider settingsConfig={config}>
          <div />
        </AppSettingsProvider>
      ));

      expect(wrapper).toMatchSnapshot();
    });
  });
});
