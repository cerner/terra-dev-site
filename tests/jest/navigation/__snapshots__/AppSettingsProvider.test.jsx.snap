// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContextProvider Snapshots should render with default config specified 1`] = `
<ContextProvider
  value={
    Object {
      "currentDirection": "rtl",
      "currentLocale": "es",
      "currentTheme": "my theme",
      "currentThemeClassName": undefined,
      "directions": Array [
        "ltr",
        "rtl",
      ],
      "locales": Array [
        "en",
        "es",
        "de",
      ],
      "onUpdate": [Function],
      "themes": Array [
        "clinical-lowlight-theme",
        "orion-fusion-theme",
      ],
    }
  }
>
  <div />
</ContextProvider>
`;

exports[`ThemeContextProvider Snapshots should render with no config 1`] = `
<ContextProvider
  value={
    Object {
      "currentDirection": "ltr",
      "currentLocale": "en",
      "currentTheme": "orion-fusion-theme",
      "currentThemeClassName": undefined,
      "directions": Array [
        "ltr",
        "rtl",
      ],
      "locales": Array [
        "en",
        "es",
        "de",
      ],
      "onUpdate": [Function],
      "themes": Array [
        "clinical-lowlight-theme",
        "orion-fusion-theme",
      ],
    }
  }
>
  <div />
</ContextProvider>
`;
